src/main/java/filescanner/SizeAction.java

```diff
@@ -0,0 +1,24 @@
+package filescanner;
+
+import org.apache.tools.ant.DirectoryScanner;
+
+import java.util.concurrent.RecursiveAction;
+
+public class SizeAction extends RecursiveAction {
+
+    private DirectoryScanner directoryScanner;
+
+    public SizeAction(DirectoryScanner directoryScanner) {
+        this.directoryScanner = directoryScanner;
+    }
+
+    @Override
+    protected void compute() {
+        int filesSize = 0;
+        for (String fileName : directoryScanner.getIncludedFiles()) {
+            filesSize += directoryScanner.getResource(fileName).getSize();
```

Comment:
are you able to write a unit test to check that it's work fine with current implementation?)

Reply:
Done.